src/algebra/group.lean

```diff
@@ -592,22 +592,44 @@ calc is_conj a 1 ↔ is_conj 1 a : ⟨is_conj_symm, is_conj_symm⟩
 
 end is_conj
 
-@[to_additive is_add_hom]
-def is_mul_hom {α β : Type*} [has_mul α] [has_mul β] (f : α → β) : Prop :=
-∀ x y, f (x * y) = f x * f y
+class is_mul_hom {α β : Type*} [has_mul α] [has_mul β] (f : α → β) : Prop :=
+(map_mul : ∀ {x y}, f (x * y) = f x * f y)
+
+class is_add_hom {α β : Type*} [has_add α] [has_add β] (f : α → β) : Prop :=
+(map_add : ∀ {x y}, f (x + y) = f x + f y)
+attribute [to_additive is_add_hom] is_mul_hom
```

Comment:
I guess you need more of these lines here.
Try `#print prefix is_mul_hom` and `#print prefix is_add_hom`. For each line in the output of the first command you need a matching line in the output of the second.

Reply:
Oh thanks! This whole attribute thing is slowly but surely making sense.